Model/Checkout/PlaceOrder.php

```diff
@@ -238,15 +238,17 @@ public function createOrder(Quote $quote, string $amwalOrderId, string $refId):
         $this->logDebug(sprintf('Submitting quote with ID %s', $quote->getId()));
         $order = $this->getOrderByAmwalOrderId($amwalOrderId);
 
-        if ($order->getEntityId()) {
-            if ($order->getState() == Order::STATE_PENDING_PAYMENT) {
-                $this->orderRepository->delete($order);
-                $order = $this->quoteManagement->submit($quote);
-            }
-        }else{
-            $order = $this->quoteManagement->submit($quote);
+        if ($order && $order->getState() !== Order::STATE_PROCESSING) {
```

Comment:
If an order exists and the state is processing, the code will continue creating another order with the same id. It should raise an exception instead.
```suggestion
        if ($order) {
            if( $order->getState() !== Order::STATE_PENDING_PAYMENT) {
                throw new RuntimeException('Found an existing order with same transacation Id with non pending payment state');
            }
```

Reply:
We can separate these flows, but is this an actual case which occurs (processing order that is being submitted again)?

Reply:
I am not sure. Just trying to be as defensive as possible. Always asserting what normal conditions should be like and raising error otherwise.

Reply:
@Skullsneeze can you please update this one too?